wordpress-mobile/WordPress-iOS
# Removes the PageListCell protocol.

Fixes #5236 

**Description:**
This PR removes the `PageListCell` protocol, and instead opts to create a base class that fulfills the same purpose in a cleaner manner.  The rationale behind this change is that protocols should NOT be used to implement common behaviour behind two classes that could instead share a common ancestor.

**Changes:**
- Removes `PageListCell` protocol.
- Adds `BasePageListCell` base class.
- Renames `PageListTableViewCellDelegate` to `BasePageListCellDelegate`.

**Future changes:**
- Remove protocol `BasePageListCellDelegate`.  [Ticket here](https://github.com/wordpress-mobile/WordPress-iOS/issues/5249).
- Remove protocol `WPPostContentViewProvider`.  [Ticket here](https://github.com/wordpress-mobile/WordPress-iOS/issues/5250)

**To test:**
1. Go to the list of pages for a blog.
2. Publish a page.
3. Make sure the page shows properly in the list of published pages.
4. Trash the page, make sure it's replaced by an "Undo" cell.
5. Undo the operation and make sure the post comes back.
6. Trash again.
7. Change the filter to "Trash".
8. Make sure the post shows there.

Also make sure these steps are reflected properly in the web interface.

Needs review: @jleandroperez
